crates/bevy_ecs/src/system/system_param.rs

```diff
@@ -210,43 +210,75 @@ fn assert_component_access_compatibility(
            query_type, filter_type, system_name, accesses);
 }
 
-/// Use to safely access and interact with up to 8 mutually exclusive [`SystemParam`]s such as
-/// two queries that reference the same mutable data  or a mutable query and acess to the ecs world.
+/// A collecton of potentially conflicting [`SystemParam`]s allowed by disjoint access.
 ///
-/// Access to each individual [`SystemParam`] by using the funcions ```p0()```, ```p1()```, ..., ```p7()```. This ensures that
-/// there's either only one mutable reference to a parameter at a time or an infinite* number of inmutable references.
-/// # Example
-/// ```
-/// use bevy_ecs::prelude::*;
+/// Allows to safely access and interact with up to 8 mutually exclusive [`SystemParam`]s such as
+/// two queries that reference the same mutable data  or a mutable query and acess to the ecs world.
 ///
-/// #[derive(Component)]
-/// struct Health(u32);
+/// Access to each individual [`SystemParam`] by using the functions `p0()`, `p1()`, ..., `p7()`. This ensures that
+/// there's either only one mutable reference to a parameter at a time or any number of immutable references.
 ///
-/// #[derive(Component)]
-/// struct Enemy;
+/// # Example
 ///
-/// //Given the following system
+/// ```
+/// # use bevy_ecs::prelude::*;
+/// #
+/// # #[derive(Component)]
+/// # struct Health;
+/// #
+/// # #[derive(Component)]
+/// # struct Enemy;
+/// #
+/// # #[derive(Component)]
+/// # struct Ally;
+/// #
+/// // Given the following system
 /// fn fancy_system(
 ///     mut set: ParamSet<(
 ///         Query<&mut Health, With<Enemy>>,
-///         Query<&mut Health, Without<Enemy>>,
+///         Query<&mut Health, With<Ally>>,
 ///         &World
 ///     )>
 /// ) {
-///     //To access the param set's elements do as follows:
+///     // To access the ParamSet's elements according to the order they're defined, use the methods p0(), p1(), ..., p7().
```

Comment:
If you accept the suggestion above, remember to also adjust or remove this comment to remove duplicated information.

You can change it to something like: “This will access the first `SystemParam`”, and so on for the usage of the second and third `SystemParam`s.